backend-plugin/src/main/java/com/redhat/jenkins/nodesharingbackend/Pool.java

```diff
@@ -107,7 +113,26 @@
         return property;
     }
 
-    public @CheckForNull StandardUsernamePasswordCredentials getCredential() {
+    public @CheckForNull UsernamePasswordCredentials getExecutorCredential(ExecutorJenkins executor) {
+        if(executor.getCredentialId() != null && !executor.getCredentialId().isEmpty()) {
```

Comment:
Nit: If ```getCreedentialId()``` can return ```NULL``` FindBugs will claim because second call of it can return ```NULL``` too -> possible NPE.